qa/os/src/test/java/org/elasticsearch/packaging/test/PackagesSecurityAutoConfigurationTests.java

```diff
@@ -0,0 +1,168 @@
+        verifySecurityAutoConfigured(installation);
+        assertNotNull(installation.getElasticPassword());
+    }
+
+    public void test20SecurityNotAutoConfiguredOnReInstallation() throws Exception {
+        // we are testing force upgrading in the current version
+        // In such a case, security remains configured from the initial installation, we don't run it again.
+        Optional<String> autoConfigDir = getAutoConfigPathDir(installation);
+        installation = Packages.forceUpgradePackage(sh, distribution);
+        assertInstalled(distribution);
+        verifyPackageInstallation(installation, distribution, sh);
+        verifySecurityAutoConfigured(installation);
+        // Since we did not auto-configure the second time, the directory name should be the same
+        assertThat(autoConfigDir.isPresent(), is(true));
+        assertThat(getAutoConfigPathDir(installation).isPresent(), is(true));
+        assertThat(getAutoConfigPathDir(installation).get(), equalTo(autoConfigDir.get()));
+    }
+
+    public void test30SecurityNotAutoConfiguredWhenExistingDataDir() throws Exception {
+        // This is a contrived example for packages where in a new installation, there is an
+        // existing data directory but the rest of the package tracked config files were removed
+        final Path dataPath = installation.data;
+        cleanup();
+        Files.createDirectory(dataPath);
+        append(dataPath.resolve("foo"), "some data");
+        installation = installPackage(sh, distribution(), existingSecurityConfiguration());
+        verifySecurityNotAutoConfigured(installation);
+    }
+
+    public void test40SecurityNotAutoConfiguredWhenExistingKeystoreUnknownPassword() throws Exception {
+        // This is a contrived example for packages where in a new installation, there is an
+        // existing elasticsearch.keystore file within $ES_PATH_CONF and it's password-protected
+        final Installation.Executables bin = installation.executables();
+        bin.keystoreTool.run("passwd", "some_password\nsome_password\n");
+        final Path tempDir = createTempDir("existing-keystore-config");
+        final Path confPath = installation.config;
+        Files.copy(
+            confPath.resolve("elasticsearch.keystore"),
+            tempDir.resolve("elasticsearch.keystore"),
+            StandardCopyOption.COPY_ATTRIBUTES
+        );
+        cleanup();
+        Files.createDirectory(confPath);
+        Files.copy(
+            tempDir.resolve("elasticsearch.keystore"),
+            confPath.resolve("elasticsearch.keystore"),
+            StandardCopyOption.COPY_ATTRIBUTES
+        );
+        installation = installPackage(sh, distribution(), errorOutput());
+        List<String> configLines = Files.readAllLines(installation.config("elasticsearch.yml"));
+        assertThat(configLines, not(hasItem("# have been automatically generated in order to configure Security.               #")));
```

Comment:
This might be more robust?
```suggestion
        assertThat(configLines, not(hasItem(containsString("have been automatically generated in order to configure Security."))));
```